### PR TITLE
Many-to-many-relations: support multi-file-upload

### PR DESCRIPTION
**Pull request by BlackbitDevs**
For (advanced) many-to-many relations with assets being allowed you can directly upload assets from the relation view:
Bildschirmfoto 2023-06-22 um 16 09 09

Currently this only supports single-file uploads. The same applies under assets context menu for "Upload (Compatibility mode)". With this PR all those places now support multi-file upload.

### CODE REVIEW

**Comment by robertSt7** [comment]
@BlackbitDevs Could you please replace `var` with `let` and `const`. Thank you

File: public/js/pimcore/helpers.js [hunk 1]
```diff
@@ -1035,34 +1035,100 @@ pimcore.helpers.uploadDialog = function (url, filename, success, failure, descri
...
         buttonConfig: {
             iconCls: 'pimcore_icon_upload'
         },
         listeners: {
             change: function (fileUploadField) {
-                if(fileUploadField.fileInputEl.dom.files[0].size > pimcore.settings["upload_max_filesize"]) {
-                    pimcore.helpers.showNotification(t("error"), t("file_is_bigger_that_upload_limit") + " " + fileUploadField.fileInputEl.dom.files[0].name, "error");
-                    return;
-                }
+                var win = new Ext.Window({
```
Comment (kingjia90) [inline public/js/pimcore/helpers.js]: Please apply the suggested changes on https://github.com/pimcore/admin-ui-classic-bundle/pull/137#issuecomment-1917028900

The rest LGTM, Thank you in advance

**Review by kingjia90 — CHANGES_REQUESTED** [review]

**Comment by kingjia90** [comment]
One minor graphical thing is that the popup that appears when clicking on the upload button, it doesn't hint that you can select one or more files 
image

File: public/js/pimcore/helpers.js (hunk 1 above)
Comment (BlackbitDevs) [reply public/js/pimcore/helpers.js]: @kingjia90 Done. Sorry, was several weeks on sick-leave.

**Comment by BlackbitDevs** [comment]
> One minor graphical thing is that the popup that appears when clicking on the upload button, it doesn't hint that you can select one or more files

Actually people will assume this as they know it from other applications, won't they?
Of course we could change the placeholder to "Select files" - this should include one or multiple. WDYT?

**Comment by kingjia90** [comment]
Yes, that's right, the text change would be fine to me

**Comment by BlackbitDevs** [comment]
Changed to "Select files" - the non-english translations are still maintained outside of github, right?

**Comment by kingjia90** [comment]
Yes, but i would use a new translation key instead.

File: public/js/pimcore/helpers.js [hunk 2]
```diff
@@ -1035,34 +1035,100 @@ pimcore.helpers.uploadDialog = function (url, filename, success, failure, descri
         emptyText: t("select_a_file"),
```
Comment (kingjia90) [inline public/js/pimcore/helpers.js]:
```suggestion
        emptyText: t("select_files"),
```

File: public/js/pimcore/helpers.js [hunk 3]
```diff
@@ -1035,34 +1035,100 @@ pimcore.helpers.uploadDialog = function (url, filename, success, failure, descri
...
+                let activeUploads = 0;
+                let filesCount = fileUploadField.fileInputEl.dom.files.length;
+
+                Ext.each(fileUploadField.fileInputEl.dom.files, function (file) {
+                    if (file.size > pimcore.settings["upload_max_filesize"]) {
+                        pimcore.helpers.showNotification(t("error"), t("file_is_bigger_that_upload_limit") + " " + file.name, "error");
+                        return;
+                    }
+
+                    var pbar = new Ext.ProgressBar({
```
Comment (kingjia90) [inline public/js/pimcore/helpers.js]:
```suggestion
                    let pbar = new Ext.ProgressBar({
```

File: public/js/pimcore/helpers.js [hunk 4]
```diff
@@ -1035,34 +1035,100 @@ pimcore.helpers.uploadDialog = function (url, filename, success, failure, descri
...
+                    data.append("filename", file.name);
+                    data.append("csrfToken", pimcore.settings['csrfToken']);
+
+                    let request = new XMLHttpRequest();
+                    let res = {
+                        'response': request
+                    };
+
+                    let successWrapper = function (ev) {
+                        var data = JSON.parse(request.responseText);
```
Comment (kingjia90) [inline public/js/pimcore/helpers.js]:
```suggestion
                        const data = JSON.parse(request.responseText);
```

File: public/js/pimcore/helpers.js [hunk 5]
```diff
@@ -1035,34 +1035,100 @@ pimcore.helpers.uploadDialog = function (url, filename, success, failure, descri
...
+                    this.activeUploads--;
+                    win.remove(pbar);
+
+                    if(this.activeUploads < 1) {
+                        win.close();
                     }
+                }.bind(this);
+
+                let activeUploads = 0;
+                let filesCount = fileUploadField.fileInputEl.dom.files.length;
```
Comment (kingjia90) [inline public/js/pimcore/helpers.js]:
```suggestion
                const filesCount = fileUploadField.fileInputEl.dom.files.length;
```

File: public/js/pimcore/helpers.js [hunk 6]
```diff
@@ -1035,34 +1035,100 @@ pimcore.helpers.uploadDialog = function (url, filename, success, failure, descri
...
         buttonConfig: {
             iconCls: 'pimcore_icon_upload'
         },
         listeners: {
             change: function (fileUploadField) {
-                if(fileUploadField.fileInputEl.dom.files[0].size > pimcore.settings["upload_max_filesize"]) {
-                    pimcore.helpers.showNotification(t("error"), t("file_is_bigger_that_upload_limit") + " " + fileUploadField.fileInputEl.dom.files[0].name, "error");
-                    return;
-                }
+                let win = new Ext.Window({
```
Comment (kingjia90) [inline public/js/pimcore/helpers.js]:
```suggestion
                const win = new Ext.Window({
```

File: public/js/pimcore/helpers.js [hunk 7]
```diff
@@ -1035,34 +1035,100 @@ pimcore.helpers.uploadDialog = function (url, filename, success, failure, descri
...
+                        width:465,
+                        text: file.name,
+                        style: "margin-bottom: 5px"
+                    });
+
+                    win.add(pbar);
+                    win.updateLayout();
+
+                    activeUploads++;
+                    let percentComplete = activeUploads / filesCount;
```
Comment (kingjia90) [inline public/js/pimcore/helpers.js]:
```suggestion
                    const percentComplete = activeUploads / filesCount;
```

File: translations/admin.en.yaml [hunk 8]
```diff
@@ -139,7 +139,7 @@ this_element_cannot_be_edited: 'This element cannot be edited'
 details: Details
 cannot_save_object_please_try_to_edit_the_object_in_detail_view: '<b>Cannot save object!</b><br />Please try to edit the object in the detail view.'
 size: Size
-select_a_file: 'Select a file'
+select_a_file: 'Select files'
```
Comment (kingjia90) [inline translations/admin.en.yaml]:
```suggestion
select_a_file: 'Select a file'
select_files: 'Select files'
```

**Review by kingjia90 — CHANGES_REQUESTED** [review]

**Pull request merged**